### PR TITLE
Use BeaconDB as default instead of MLS

### PR DESCRIPTION
**Pull request by RedAuburn**
Should I also remove the MLS depreciation warning, or just modify it?:

fixes https://github.com/mjaakko/NeoStumbler/issues/189

### CODE REVIEW

File: app/src/main/java/xyz/malkki/neostumbler/geosubmit/GeosubmitParams.kt [hunk 1]
```diff
@@ -10,8 +10,7 @@ data class GeosubmitParams(
     val apiKey: String?
 ) {
     companion object {
-        //NOTE: MLS does not accept data submissions anymore, this is used as a default just to show an example
-        const val DEFAULT_BASE_URL = "https://location.services.mozilla.com"
+        const val DEFAULT_BASE_URL = "https://beacondb.net"
```
Comment (Mikaela) [inline app/src/main/java/xyz/malkki/neostumbler/geosubmit/GeosubmitParams.kt:13]: @joelkoen requested on Matrix that defaults would use the api subdomain, which while functionally identical, will allow scaling or something if necessary, so I think changing the default here should also use it.

```suggestion
        const val DEFAULT_BASE_URL = "https://api.beacondb.net"
```

**Review by Mikaela — CHANGES_REQUESTED** [review]

File: app/src/main/java/xyz/malkki/neostumbler/geosubmit/GeosubmitParams.kt (hunk 1 above)
Comment (RedAuburn) [reply app/src/main/java/xyz/malkki/neostumbler/geosubmit/GeosubmitParams.kt]: ah good catch, thanks!

**Comment by mjaakko** [comment]
I'm planning on doing this as discussed in here: https://github.com/mjaakko/NeoStumbler/issues/189#issuecomment-2283043476

Forcing the user to choose the endpoint before sending any data has the advantages that:
1. the user understands where their data is sent and can review relevant terms of use etc.
2. NeoStumbler won't be tightly coupled with a single service (relevant with #79 and #124)

At least there would have to be a warning if the default endpoint was changed

**Comment by RedAuburn** [comment]
> I'm planning on doing this as discussed in here: [#189 (comment)](https://github.com/mjaakko/NeoStumbler/issues/189#issuecomment-2283043476)

ah ok :+1:

**Pull request closed without merging**